packages/oscal-react-library/src/components/OSCALCatalogGroup.js

```diff
@@ -64,32 +86,71 @@ function CollapsibleListItem(props) {
       isSetListItemNavigatedTo(true);
       return;
     }
-    // Ensure fragment exists and split by groupings
-    if (!urlFragment) {
-      return;
-    }
-    // Find control list state and open collapsible item
-    const currentControl = fragmentSuffix.split("/")[0];
-    if (currentControl === control?.id) {
+
+    if (isMatchingListItem(urlFragment, previousHandledFragment, fragmentSuffix, listId)) {
       setIsOpen(true);
-      const elementWithFragment = document.getElementById(control.id);
-      elementWithFragment?.scrollIntoView?.({ behavior: "smooth" });
+
+      if (fragmentSuffix.split("/").length <= 1) {
+        const elementWithFragment = document.getElementById(urlFragment);
+        elementWithFragment?.scrollIntoView?.({ behavior: "smooth" });
+        setPreviousHandledFragment(urlFragment);
+      }
     }
-  }, [urlFragment, fragmentSuffix, listItemOpened, isSetListItemNavigatedTo, control?.id]);
+  }, [
+    urlFragment,
+    fragmentSuffix,
+    listItemOpened,
+    isSetListItemNavigatedTo,
+    listId,
+    previousHandledFragment,
+    setPreviousHandledFragment,
+  ]);
 
   return (
     <StyledListItemPaper>
       <StyledListItem onClick={handleClick}>
         <ListItemText primary={itemText} />
         {isOpen ? <ExpandLess /> : <ExpandMore />}
       </StyledListItem>
-      <Collapse in={isOpen} timeout="auto" onEntered={() => setListItemOpened(true)} unmountOnExit>
+      <Collapse
+        in={isOpen}
+        timeout="auto"
+        onEntered={() => !group && setListItemOpened(true)}
+        unmountOnExit
+      >
         <StyledControlDescriptionWrapper>{children}</StyledControlDescriptionWrapper>
       </Collapse>
     </StyledListItemPaper>
   );
 }
 
+function WithdrawnListItem(props) {
```

Comment:
Same thing here. `jsdoc` will make switching to TypeScript a lot less painful.

Reply:
This goes with the question as stated previously on JSDoc support. I consider this to be like an OSCAL component, just not under the OSCAL name, similar to `CollapsibleListItem`.